Assets/_Laz/Scripts/Managers/CheckPointManager.cs

```diff
@@ -1,28 +1,70 @@
-using System.Collections.Generic;
+using PerigonGames;
 using UnityEngine;
 
 namespace Laz
 {
     public class CheckPointManager : MonoBehaviour
     {
-        [SerializeField] private List<Transform> _checkPoints = null;
-        public Transform LastCheckPoint { get; private set; } = null;
+        [SerializeField] private Checkpoint[] _checkpoints;
+        private Checkpoint _activeCheckpoint = null;
+        private Vector3 _initialCheckPointPosition = Vector3.zero;
+        private LazPlayer _laz = null;
 
-        public void Initialize()
+        public void Initialize(LazPlayer laz)
         {
-            UpdateCheckPoint();
+            if (_checkpoints.IsNullOrEmpty())
+            {
+                Debug.LogError("No Checkpoints Found!");
+                return;
+            }
+
+            _laz = laz;
+            
+            for (int i = 0; i < _checkpoints.Length; i++)
+            {
+                Checkpoint checkpoint = _checkpoints[i];
+
+                if (checkpoint == null)
+                {
+                    Debug.LogError("Null Checkpoint");
```

Comment:
This can be more specific as well. "There are Null Checkpoints within the CheckpointManager"